scripts/devops_tasks/tox_harness.py

```diff
@@ -274,11 +274,66 @@ def replace_dev_reqs(file, pkg_root):
         f.write("\n".join(adjusted_req_lines))
 
 
+def collect_log_files(working_dir):
+    logging.info("Collecting log files from {}".format(working_dir))
+    package = working_dir.split('/')[-1]
+    # collect all the log files into one place for publishing in case of tox failure
+    test_envs = ["whl", "sdist", "depends", "latestdependency", "mindependency"]
+
+    log_directory = os.path.join(
+        root_dir, "_tox_logs"
+    )
+
+    try:
+        os.mkdir(log_directory)
+    except OSError:
+        logging.info("'{}' directory already exists".format(log_directory))
+
+    log_directory = os.path.join(
+        log_directory, package
+    )
+
+    try:
+        os.mkdir(log_directory)
+    except OSError:
+        logging.info("'{}' directory already exists".format(log_directory))
+
+    log_directory = os.path.join(
+        log_directory, sys.version.split()[0]
```

Comment:
So the final result log directory looks like:

~~`$(Build.SourcesDirectory)/_tox_logs/<packagename>/3/*.txt` If I understand this correctly.~~

This is totally fine. `sys.version.split()` returned something I did not expect.

Reply:
Just as an aside. I would also deliberately include a failing assertion in a random test just to double check the artifact publishing. If you have already done that 👍 good by me.